gnark/gnark-jni/gnark-eip-2537.go

```diff
@@ -681,33 +514,44 @@ func eip2537blsMapFpToG1(javaInputBuf, javaOutputBuf, javaErrorBuf *C.char, cInp
 	inputLen := int(cInputLen)
 	errorLen := int(cOutputLen)
 
-	// Convert error C pointers to Go slices
+	// Convert C pointer to error buffer into a Go slice
 	errorBuf := castBuffer(javaErrorBuf, errorLen)
 
+	// Validate input length and convert C pointer to input buffer into a Go slice
 	if inputLen != (EIP2537PreallocateForFp) {
 		copy(errorBuf, "invalid input parameters, invalid input length for Fp to G1 to curve mapping\x00")
 		return 1
 	}
-
-	// Convert input C pointers to Go slice
 	input := (*[EIP2537PreallocateForFp]byte)(unsafe.Pointer(javaInputBuf))[:inputLen:inputLen]
 
-	if !isZero(input[:16]) {
-		copy(errorBuf, ErrMalformedPointPadding.Error())
+	// Map Fp field element to a G1 point
+	result, err := _blsMapFpToG1(input)
+	if err != nil {
+		copy(errorBuf, err.Error())
```

Comment:
Seems like we could add a dryError() similar to gnark-eip-196.go which could copy and add the string termination. (though eip-196 dry error does not add string terminator either)

Reply:
I was planning to do this change in a separate PR, so that this PR mainly removed extra files